rust/common-javascript-interpreter/src/lib.rs

```diff
@@ -6,40 +6,50 @@
 //! do in a AOT-compiled scenario. Notably: this package is designed to be
 //! compiled as a Wasm Component exporting the `common:script` target. This
 //! enables us to evaluate `common:module` JavaScript within a Wasm sandbox.
-
+#[cfg(all(target_arch = "wasm32", target_os = "wasi"))]
 #[allow(warnings)]
 mod bindings;
 
+#[cfg(all(target_arch = "wasm32", target_os = "wasi"))]
 mod data;
+#[cfg(all(target_arch = "wasm32", target_os = "wasi"))]
 mod io;
+#[cfg(all(target_arch = "wasm32", target_os = "wasi"))]
 mod module;
 
-use bindings::Guest;
-// use io::read_script;
-use module::Module;
+#[cfg(all(target_arch = "wasm32", target_os = "wasi"))]
```

Comment:
It would probably be cleaner to move the `Guest` impl out into its own file so that we can reduce ~4 compiler conditionals to 1 or 2.

Reply:
SG, can do that on the next change to the interpreter